src/nvim/normal.c

```diff
@@ -4388,10 +4386,10 @@ static void nv_ident(cmdarg_T *cap)
       STRCAT(buf, "\\>");
     /* put pattern in search history */
     init_history();
-    add_to_history(HIST_SEARCH, buf, true, NUL);
-    normal_search(cap, cmdchar == '*' ? '/' : '?', buf, 0);
+    add_to_history(HIST_SEARCH, (char_u *)buf, true, NUL);
```

Comment:
The amount of bytes is the same (and the signedness of char is platform depedent, it's a weird type). It should be absolutely fine. The only thing that might differ is the generated instructions: wether they are signed or unsigned adds/subs/cmps/...

Said another way, it's not as dangerous as casting (uint64_t *) to (uin32_t *).